### PR TITLE
OF-2559: EncryptionFactory improvements for Netty

### PR DESCRIPTION
**Pull request by guusdk**
Please review the individual commit messages. The first one fixes a bug in the new Netty implementation, the second one is more of a cleanup.

### CODE REVIEW

**Comment by guusdk** [comment]
I did not check if there are methods that were only used by MINA and could potentially be removed.

File: xmppserver/src/main/java/org/jivesoftware/openfire/spi/EncryptionArtifactFactory.java [hunk 1]
```diff
@@ -367,6 +367,7 @@ public SslContext createServerModeSslContext(boolean directTLS) throws Unrecover
                 break;
         }
 
+        builder.trustManager(getTrustManagers()[0]); // The existing implementation never returns more than one trust manager.
```
Comment (viv) [inline xmppserver/src/main/java/org/jivesoftware/openfire/spi/EncryptionArtifactFactory.java:385]: Should we also set the keyManager here?
`builder.keyManager(getKeyManagerFactory());`

**Review by viv — APPROVED** [review]

**Review by AlexGidman — APPROVED** [review]

**Comment by AlexGidman** [comment]
S2S docker tests still working as before with TLSv1.3

File: xmppserver/src/main/java/org/jivesoftware/openfire/spi/EncryptionArtifactFactory.java (hunk 1 above)
Comment (guusdk) [reply xmppserver/src/main/java/org/jivesoftware/openfire/spi/EncryptionArtifactFactory.java:385]: I don't think so, as the `keyManagerFactory` is already passed in de factory method of the `SslContextBuilder` isntance, a few lines higher.

**Pull request merged**